app/main/system/serializers.ts

```diff
@@ -0,0 +1,19 @@
+import * as f from "fp-ts/lib/function"
+import { Lifecycle, pureComponent } from "./lifecycle"
+
+export type Serializer = {
+  encode(value: any): Promise<string>;
+  decode(text: string): Promise<any>;
+}
+
+export const json: Lifecycle<Serializer, void> = pureComponent(f.constant({
+  encode: wrapPromise((v: any) => JSON.stringify(v)),
+  decode: wrapPromise((t: string) => JSON.parse(t))
+}))
```

Comment:
Anonymous functions can be easily wrapped in a promise in place by simply using `async`:
```ts
encode: async (value: JsonSerializable) => JSON.stringify(value)
```